src/AgentContainer.cpp

```diff
@@ -674,6 +674,12 @@ void AgentContainer::infectAgents ()
                             latent_period_ptr[i] = amrex::RandomNormal(lparm->latent_length_mean, lparm->latent_length_std, engine);
                             infectious_period_ptr[i] = amrex::RandomNormal(lparm->infectious_length_mean, lparm->infectious_length_std, engine);
                             incubation_period_ptr[i] = amrex::RandomNormal(lparm->incubation_length_mean, lparm->incubation_length_std, engine);
+                            if (latent_period_ptr[i] < 0) { latent_period_ptr[i] = 0.0_rt; }
+                            if (infectious_period_ptr[i] < 0) { infectious_period_ptr[i] = 0.0_rt; }
+                            if (incubation_period_ptr[i] < 0) { incubation_period_ptr[i] = 0.0_rt; }
+                            if (latent_period_ptr[i] > (infectious_period_ptr[i]+incubation_period_ptr[i])) {
+                                latent_period_ptr[i] = std::floor(infectious_period_ptr[i]+incubation_period_ptr[i]);
+                            }
```

Comment:
> Hi Debo, you were right that some agents had longer incubation period than the sum of their latent and infectious periods. The assertion error went away with your new code. However, I think the inconsistency among these times still remains and may cause other errors in the future. I observed that some agents even got a negative latent period... Should we enforce some rules when we generate random numbers for these periods here?
> 
> https://github.com/AMReX-Codes/ExaEpi/blob/5d83e0a3c351375100bbabad9faaba2f234defc2/src/AgentContainer.cpp#L674
> 
> E.g., we can keep rolling the dices until all numbers make sense.

@tannguyen153 Yes, that's a good point. Check out what I have done now to fix this:
- If any of these periods are negative, then make them `0.0`.
- If the latent period is greater than the sum of the incubation and infectious periods, then set it to the floor of that sum.

Reply:
Thanks Debo. Just a quick question. Your new code enforces latent to be less than incubation+infectious. Shouldn't it be incubation<(latent+infectious)? Also, I think latent has to be shorter than incubation.

Reply:
@tannguyen153 You are right; I fixed it. Thanks for catching this!

Reply:
good question. I thought about it again. I think read some where that latent and infectious periods can overlap in different ways depending on the disease. The general case is that hosts become infectious first (latent) and then show symptoms (incubation). However, some hosts can have symptoms first before becoming infectious. We can ask the Sandia team about this when we meet next week.